yajsapi/executor/strategy.ts

```diff
@@ -0,0 +1,101 @@
+import { Activity } from "../props";
+import { DemandBuilder } from "../props/builder";
+import {
+  BillingScheme,
+  ComLinear,
+  Counter,
+  PriceModel,
+  PRICE_MODEL,
+} from "../props/com";
+import { OfferProposal } from "../rest/market";
+import { applyMixins } from "../utils";
+
+export const SCORE_NEUTRAL: number = 0.0;
+export const SCORE_REJECTED: number = -1.0;
+export const SCORE_TRUSTED: number = 100.0;
+
+export const CFF_DEFAULT_PRICE_FOR_COUNTER: Map<Counter, number> = new Map([
+  [Counter.TIME, parseFloat("0.002")],
+  [Counter.CPU, parseFloat("0.002") * 10],
+]);
+
+export class MarketStrategy {
+  /*Abstract market strategy*/
+
+  async decorate_demand(demand: DemandBuilder): Promise<void> {}
+
+  async score_offer(offer: OfferProposal): Promise<Number> {
+    return SCORE_REJECTED;
+  }
+}
+
+interface MarketGeneral extends MarketStrategy, Object {}
+class MarketGeneral {}
+
+applyMixins(MarketGeneral, [MarketStrategy, Object]);
+
+export class DummyMS extends MarketGeneral {
+  max_for_counter: Map<Counter, Number> = CFF_DEFAULT_PRICE_FOR_COUNTER;
+  max_fixed: Number = parseFloat("0.05");
+  _activity?: Activity;
+
+  async decorate_demand(demand: DemandBuilder): Promise<void> {
+    demand.ensure(`(${PRICE_MODEL}=${PriceModel.LINEAR})`);
+    this._activity = new Activity().from_properties(demand._properties);
+  }
+
+  async score_offer(offer: OfferProposal): Promise<Number> {
+    const linear: ComLinear = new ComLinear().from_properties(offer.props());
+
+    if (linear.scheme.value != BillingScheme.PAYU) {
+      return SCORE_REJECTED;
+    }
+
+    if (linear.fixed_price > this.max_fixed) return SCORE_REJECTED;
```

Comment:
This is not related to renaming, but it's risky to reject offers with an arbitrarily chosen `0.05` initial price. Is this value high or low?

Reply:
This seems important, I suggest create an issue for it.